extensions/slack-status/src/slack.ts

```diff
@@ -9,13 +10,45 @@ export function useSlack() {
   return webClient;
 }
 
-export function useSlackProfile() {
+export function useSlackProfileAndDndInfo(slackAuth) {
   const slack = useSlack();
+
   return usePromise(async () => {
-    const response = await slack.users.profile.get();
-    if (!response.ok) {
+    const profileResponse = await slack.users.profile.get();
+    if (!profileResponse.ok) {
       throw Error("Failed to fetch profile");
     }
-    return response.profile;
+
+    let dndResponse;
+    let dndCalled = false;
+
+    while (!dndCalled) {
+      try {
+        dndResponse = await slack.dnd.info();
+        dndCalled = true;
+        if (!dndResponse.ok) {
+          throw Error("Failed to fetch DND info");
+        }
+      } catch (error) {
+        // We added new scopes for DND. Old installs don't automatically upgrade scopes.
```

Comment:
As discussed in https://raycastcommunity.slack.com/archives/C02HEMAF2SJ/p1734186555949399.